includes/class-wc-payments-order-service.php

```diff
@@ -794,4 +795,54 @@ private function update_order_status( $order, $order_status, $intent_id = '' ) {
 			Logger::log( 'Error when updating status for order ' . $order->get_id() . ': ' . $e->getMessage() );
 		}
 	}
+
+	/**
+	 * Return terminal intent payment method array based on payment methods request.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @param array           $default_value - default value.
+	 *
+	 * @return array|null
+	 * @throws \Exception
+	 */
+	public function get_terminal_intent_payment_method( $request, array $default_value = [ Payment_Method::CARD_PRESENT ] ) :array {
+		$payment_methods = $request->get_param( 'payment_methods' );
+		if ( null === $payment_methods ) {
+			return $default_value;
+		}
+
+		if ( ! is_array( $payment_methods ) ) {
+			throw new \Exception( 'Invalid param \'payment_methods\'!' );
+		}
+
+		foreach ( $payment_methods as $value ) {
+			if ( ! in_array( $value, Payment_Method::IPP_ALLOWED_PAYMENT_METHODS, true ) ) {
+				throw new \Exception( 'One or more payment methods are not supported!' );
+			}
+		}
+
+		return $payment_methods;
+	}
+
+	/**
+	 * Return terminal intent capture method based on capture method request.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @param string          $default_value default value.
+	 *
+	 * @return string|null
+	 * @throws \Exception
+	 */
+	public function get_terminal_intent_capture_method( $request, string $default_value = 'manual' ) : string {
```

Comment:
This method doesn;t belong to the service (same for the previous one). Perhaps it can be moved into the controller?